pkg/proxy/userspace/proxier_test.go

```diff
@@ -124,6 +126,9 @@ func init() {
 	}
 	udpServerPort = int32(udpServerPortValue)
 	go udp.Loop()
+
+	ret := m.Run()
+	os.Exit(ret)
```

Comment:
I think the os.Exit will cause the program to exit before it does the deferred tcp.Close().